pkg/controller/operators/catalog/step_ensurer.go

```diff
@@ -116,12 +116,35 @@ func (o *StepEnsurer) EnsureServiceAccount(namespace string, sa *corev1.ServiceA
 		return
 	}
 
+	// Before UpdateServiceAccount and dereference secrets of the older SA,
+	// we capture those secrets and removed them right after update is successful. Without this,
+	// the older secrets will lay around till the operator is uninstalled.
+	// We clean up instead of attaching those older secrets to ensure that new secrets for the SA are valid.
+	preSa, getErr := o.kubeClient.KubernetesInterface().CoreV1().ServiceAccounts(namespace).Get(sa.Name,
+		metav1.GetOptions{})
+	// Returns error if no secret is found for a service account. Because according to Kubernetes,
+	// The controller loop ensures a secret with an API token exists for each service account.
+	if getErr != nil {
+		err = errorwrap.Wrapf(getErr, "error getting older secrets for service account: %s", sa.GetName())
+		return
+	}
+
 	sa.SetNamespace(namespace)
 	if _, updateErr := o.kubeClient.UpdateServiceAccount(sa); updateErr != nil {
 		err = errorwrap.Wrapf(updateErr, "error updating service account: %s", sa.GetName())
 		return
 	}
 
+	for _, secret := range preSa.Secrets {
+		foregroundDelete := metav1.DeletePropagationForeground
```

Comment:
I don't think doing a foreground type of deletion is necessary if there's nothing left referencing these secrets as dependencies, right?

Reply:
You are right. This is unnecessary but I thought it would not hurt. Do you think we should just leave it blank then? Thanks

Reply:
I believe you should have nothing passed into the DeleteOptions below.